{"path":"solr/solrj/src/test/org/apache/solr/client/solrj/io/stream/CloudAuthStreamTest#testSanityCheckAuth().mjava","commits":[{"id":"d959ec4c6f38b465c2b853271aecb516e06e7d94","date":1580403663,"type":0,"author":"Chris Hostetter","isMerge":false,"pathNew":"solr/solrj/src/test/org/apache/solr/client/solrj/io/stream/CloudAuthStreamTest#testSanityCheckAuth().mjava","pathOld":"/dev/null","sourceNew":"  /**\n   * Simple sanity checks that authentication is working the way the test expects \n   */\n  public void testSanityCheckAuth() throws Exception {\n    \n    assertEquals(\"sanity check of non authenticated query request\",\n                 401,\n                 expectThrows(SolrException.class, () -> {\n                     final long ignored = \n                       (new QueryRequest(params(\"q\", \"*:*\",\n                                                \"rows\", \"0\",\n                                                \"_trace\", \"no_auth_sanity_check\")))\n                       .process(cluster.getSolrClient(), COLLECTION_X).getResults().getNumFound();\n                   }).code());\n    \n    assertEquals(\"sanity check of update to X from write_X user\",\n                 0,\n                 (setBasicAuthCredentials(new UpdateRequest(), WRITE_X_USER)\n                  .add(sdoc(\"id\", \"1_from_write_X_user\"))\n                  .commit(cluster.getSolrClient(), COLLECTION_X)).getStatus());\n\n    assertEquals(\"sanity check of update to X from read only user\",\n                 500, // should be 403, but CloudSolrClient lies on updates for now: SOLR-14222 \n                 expectThrows(SolrException.class, () -> {\n                     final int ignored = (setBasicAuthCredentials(new UpdateRequest(), READ_ONLY_USER)\n                                          .add(sdoc(\"id\", \"2_from_read_only_user\"))\n                                          .commit(cluster.getSolrClient(), COLLECTION_X)).getStatus();\n                   }).code());\n    \n    assertEquals(\"sanity check of update to X from write_Y user\",\n                 500, // should be 403, but CloudSolrClient lies on updates for now: SOLR-14222 \n                 expectThrows(SolrException.class, () -> {\n                     final int ignored = (setBasicAuthCredentials(new UpdateRequest(), WRITE_Y_USER)\n                                          .add(sdoc(\"id\", \"3_from_write_Y_user\"))\n                                          .commit(cluster.getSolrClient(), COLLECTION_X)).getStatus();\n                   }).code());\n    \n    assertEquals(\"sanity check of update to Y from write_Y user\",\n                 0,\n                 (setBasicAuthCredentials(new UpdateRequest(), WRITE_Y_USER)\n                  .add(sdoc(\"id\", \"1_from_write_Y_user\"))\n                  .commit(cluster.getSolrClient(), COLLECTION_Y)).getStatus());\n    \n    for (String user : Arrays.asList(READ_ONLY_USER, WRITE_Y_USER, WRITE_X_USER)) {\n      for (String collection : Arrays.asList(COLLECTION_X, COLLECTION_Y)) {\n        assertEquals(\"sanity check: query \"+collection+\" from user: \"+user,\n                     1, countDocsInCollection(collection, user));\n      }\n    }\n  }\n\n","sourceOld":null,"bugFix":null,"bugIntro":[],"isBuggy":false,"nexts":[],"revCommit":null}],"commit2Parents":{"d959ec4c6f38b465c2b853271aecb516e06e7d94":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":[],"cd5edd1f2b162a5cfa08efd17851a07373a96817":["d959ec4c6f38b465c2b853271aecb516e06e7d94"]},"commit2Childs":{"d959ec4c6f38b465c2b853271aecb516e06e7d94":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85":["d959ec4c6f38b465c2b853271aecb516e06e7d94"],"cd5edd1f2b162a5cfa08efd17851a07373a96817":[]},"heads":["cd5edd1f2b162a5cfa08efd17851a07373a96817"],"roots":["a0e7ee9d0d12370e8d2b5ae0a23b6e687e018d85"],"pathCommit":null}